### PR TITLE
Add ETC1  builtin alpha support, separated alpha file mode.

### PR DESCRIPTION
**Pull request by halx99**

### CODE REVIEW

**Comment by halx99** [comment]
@minggo please take a look

**Comment by minggo** [comment]
@halx99 sorry, i think the PR include too many things. Could you please break it into more PRs and just do one thing in a PR. Thanks.

**Comment by halx99** [comment]
OK, I will do it.

File: tests/cpp-tests/Resources/ccs-res [hunk 1]
```diff
@@ -1 +1 @@
-Subproject commit 892e3cbdc27d24c39c62d700a81be27e593823b7
+Subproject commit fab8c569a8f27a2ff7c1150c22507d727efe5c17
```
Comment (dumganhar) [inline tests/cpp-tests/Resources/ccs-res:1]: Need to change the submodule reference?

File: tools/cocos2d-console [hunk 2]
```diff
@@ -1 +1 @@
-Subproject commit 2ceba42b89945229ac22d09c0864c86b113f3f17
+Subproject commit 04ade20dec5508d3865c89191e2f88dd21972456
```
Comment (dumganhar) [inline tools/cocos2d-console:1]: Need to change the submodule reference?

File: web [hunk 3]
```diff
@@ -1 +1 @@
-Subproject commit 99c7025c56a907fda5d3008948821e9cd0408329
+Subproject commit 2b3a2df2cf17552daf2e78fcb69b944412090d26
```
Comment (dumganhar) [inline web:1]: Need to change the submodule reference?

Comment (halx99) [inline web:1]: I'm not chagne the submodule refernece manually, I don't known what's happen.

**Comment by halx99** [comment]
This pr close, see:  https://github.com/cocos2d/cocos2d-x/pull/16118

File: cocos/renderer/CCTextureCache.h [hunk 4]
```diff
@@ -82,6 +82,9 @@ class CC_DLL TextureCache : public Ref
      */
     CC_DEPRECATED_ATTRIBUTE static void reloadAllTextures();
 
+    // ETC1 ALPHA supports.
+    static void setETC1AlphaFileEndix(const std::string& endix);
```
Comment (dumganhar) [inline cocos/renderer/CCTextureCache.h:86]: What does `Endix` stand for? Is it better to name it to `suffix`?

Comment (halx99) [inline cocos/renderer/CCTextureCache.h:86]: Yes, you are right.

**Pull request closed without merging**